web/src/components/TupleInput.vue

```diff
@@ -0,0 +1,91 @@
+<script lang="ts" setup>
+import { computed } from "vue"
+import InputElement from "@/components/InputElement.vue"
+import InputFieldComponent from "@/components/InputField.vue"
+import SubField from "@/components/SubField.vue"
+import { useInputModel } from "@/composables/useInputModel"
+import { ValidationError } from "@/domain"
+import type { InputField, ModelType } from "@/domain"
+
+interface Props {
+  ariaLabel: string
+  value?: { parent: ModelType; child: ModelType }
+  modelValue?: { parent: ModelType; child: ModelType }
+  fields: { parent: InputField; child: InputField }
+  validationError?: ValidationError
+}
+
+interface Emits {
+  (
+    event: "update:modelValue",
+    value: { parent: ModelType; child: ModelType } | undefined
+  ): void
+  (event: "input", value: Event): void
+}
+
+const props = defineProps<Props>()
+const emit = defineEmits<Emits>()
+
+const { inputValue } = useInputModel<
+  { parent: ModelType; child: ModelType },
+  Props,
+  Emits
+>(props, emit)
+
+const parentValue = computed({
+  get: () => (props.modelValue ? props.modelValue.parent : ""),
```

Comment:
I don't think we can just use empty strings here. Tuples should be generic. For example MultiTextInputs expect `string[] | undefined` as `modelValue.` 

(same for line 43)